### PR TITLE
feat(settings): Move settings under `OrganizationDetails` tree

### PR DESCRIPTION
**Pull request by billyvg**
This moves the "settings" route under `OrganizationDetails` tree. This
solves the unnecessary layout change when you go from main application into
settings (sidebar disappears and we have to refetch organization details
again). Account settings should be able to have org in context as well by
using the `lastOrganization` saved in user session. The account settings
view needs to be able to support no organization in the case of a new user
who has been invited to join an org for the first time and needs to setup
2fa before being able to join.

TODO:
- [x] Fix error handling in OrganizationDetails (sometimes we want to an error message, sometimes we do not)
- [x] Fix `componentDidUpdate` conditions (switching to account details still triggers full load)
- [ ] ....

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/components/settingsWrapper.jsx [hunk 1]
```diff
@@ -7,11 +7,20 @@ import SentryTypes from 'app/sentryTypes';
...
   color: ${p => p.theme.gray5};
-  margin: 0 auto;
+  margin-bottom: -20px; /* to account for footer margin top */
   line-height: 1;
+
+  .messages-container {
+    margin: 0;
+  }
+  .sentry-error-embed-wrapper {
+    z-index: 1005;
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/settings/components/settingsWrapper.jsx]: Should you use `theme.zIndex` here?

**Review by markstory — COMMENTED** [review]
I did some clicking around in settings and navigation was working well for me.

**Comment by billyvg** [comment]
@getsentry/growth Any issues with having the install prompt included in settings now with this change?

**Comment by billyvg** [comment]
There's a regression in the account settings area, looking into it.

**Comment by adhiraj** [comment]
Which install prompt, the black and yellow one? What causes it to appear in settings now and not before this change?

**Comment by billyvg** [comment]
@adhiraj 	Yeah the yellow/black one. Currently settings are in a completely different route tree than the main application (because account settings currently does not guarantee an org in scope), this PR moves it under the same component tree.

**Comment by adhiraj** [comment]
Yup that should be fine (as long as the link in the prompt still works)

File: src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx [hunk 2]
```diff
@@ -1,16 +1,33 @@
...
 import InlineSvg from 'app/components/inlineSvg';
-import SentryTypes from 'app/sentryTypes';
-import replaceRouterParams from 'app/utils/replaceRouterParams';
-import withLatestContext from 'app/utils/withLatestContext';
 import space from 'app/styles/space';
 
+class SettingsBackButton extends React.Component {
+  render() {
+    // if the user needs to setup 2fa as part of the org invite flow,
+    // send them back to accept the invite
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx]: I don't understand this comment in context of a render?

**Review by evanpurkhiser — APPROVED** [review]
Looking good to me if this is good to go.

File: src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/components/settingsBackButton.jsx]: It's just describing what `pendingInvite` is for

**Pull request merged**